pkg/volume/hostpath/host_path.go

```diff
@@ -142,7 +142,7 @@ func (plugin *hostPathPlugin) Recycle(pvName string, spec *volume.Spec, eventRec
 		return fmt.Errorf("spec.PersistentVolume.Spec.HostPath is nil")
 	}
 
-	pod := plugin.config.RecyclerPodTemplate
+	pod := plugin.config.RecyclerPodTemplate.DeepCopy()
```

Comment:
good catch!

Reply:
Would it be worth adding a unit test that recycles twice to catch this error?

Reply:
I will try to add unit test.

Reply:
Is hard to add unit test for fake client. I have test in real kubernetes environment.

Reply:
Sorry for the delay. What do you think of splitting this function into two pieces: 1) One that generates the recycler pod spec, and 2) one that invokes the pod. Then we can unit test the first.